### DISCUSSION TITLE
How Do I Conditionally Generate Backend

### DISCUSSION THREAD

**Discussion by AStrangwood** [gruntwork-io/knowledge-base#838] [How do I]
I have a need were i want to generate two different backends depending on if im running locally or in a CI. To be more specific i would want the `remote_state` block to use an AWS `profile` if im running locally and to use `assume_role` if in a CI. 

I cant find any way to conditionally set the vars for the `remote_state` block. I tried to conditionally set each value of `assume_role` and `profile` but that fails as you cant set `assume_role` vars to empty strings

**Comment by denis256** [comment]
Hi,
I was thinking of using `include` to include different remote_state definitions based on env variables:

```
include {
  path = get_env("CI", "false") == "true" ? "remote_state_ci.hcl": "remote_state_local.hcl"
}
```

Super simplified example in: https://github.com/denis256/terragrunt-tests/tree/master/discussion-838
